vitest.global-browser-setup.ts

```diff
@@ -7,18 +7,29 @@ export default async function setup() {
       shell: 'sh',
     });
 
-    cp.stdout?.on('data', () => {
-      // return teardown function to be called when tests finish
-      // it will kill the server
-      resolve(() => {
+    const killServer = () => {
+      if (cp.pid) {
         // https://github.com/nodejs/node/issues/2098#issuecomment-169549789
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        process.kill(-cp.pid!);
-      });
+        process.kill(-cp.pid);
+      }
+    };
+
+    cp.stdout?.on('data', () => {
+      // Return teardown function to be called when tests finish
+      // It will kill the server
+      resolve(killServer);
     });
 
     cp.on('error', (err) => {
+      // Ensure server is killed if there's an error
```

Comment:
Here are a couple more events that can be used (on parent process?) to detect non-clean exits.

Not sure which would be applicable here.

https://github.com/FuelLabs/fuels-ts/blob/bb5a123f46313f10d444d361fcfcf52e40f2430c/packages/account/src/test-utils/launchNode.ts#L284-L299

Reply:
Good that you commented on this! I analyzed killing a browser test suite mid-run and realized that it doesn't work as expected. I had to add an additional `/close-server` endpoint in https://github.com/FuelLabs/fuels-ts/pull/2378/commits/624f24b2e9f3065057a33b478fc9aa333b83c5de to make it work. The child process is run in a detached state to decouple it from the main process (to get away from polyfills) so killing it via this new endpoint was the easiest route - no shared `process` state and "I can kinda but not really kill the child process" shenanigans.